dev/com.ibm.ws.security.mp.jwt.1.2_fat/fat/src/com/ibm/ws/security/mp/jwt12/fat/sharedTests/MPJwt12MPConfigTests.java

```diff
@@ -0,0 +1,647 @@
+                                                               buildMPConfigFileContent(overrideMpConfigSettings, MpJwt12FatConstants.AUTHORIZATION,
+                                                                                        MP12ConfigSettings.CookieNotSet,
+                                                                                        mpConfigSettings.getAudience(), MpJwt12FatConstants.SIGALG_ES256));
+            setupUtils.deployRSServerMPConfigInAppUnderWebInfApp(server, MpJwt12FatConstants.GOOD_KEY_AND_ALGORITHM_IN_CONFIG_UNDER_WEB_INF_ROOT_CONTEXT,
+                                                                 buildMPConfigFileContent(overrideMpConfigSettings, MpJwt12FatConstants.AUTHORIZATION,
+                                                                                          MP12ConfigSettings.CookieNotSet,
+                                                                                          mpConfigSettings.getAudience(), MpJwt12FatConstants.SIGALG_ES256));
+
+        } catch (Exception e) {
+            Log.info(thisClass, "MPJwtAltConfig", "Hit an exception updating the war file" + e.getMessage());
+            throw e;
+        }
+
+    }
+
+    public Map<String, String> addToMap(Map<String, String> currentMap, String name, String value) throws Exception {
+
+        if (currentMap == null) {
+            currentMap = new HashMap<String, String>();
+        }
+        currentMap.put(name, value);
+        return currentMap;
+    }
+
+    public Page invokeUrlWithOtherHeader(WebClient webClient, String url, String location, String name, String value) throws Exception {
+
+        Map<String, String> header = addToMap(null, location, name + " " + value);
+
+        return actions.invokeUrlWithParametersAndHeaders(_testName, webClient, url, null, header);
+
+    }
+
+    /**
+     * All of the tests in this class follow the same flow. The differences between them are which builder they use to create a
+     * token, the config they use in the resource server
+     * and then whether they expect a failure (mainly due to a mis-match between the token and the servers config).
+     * We'll put the common steps in this method so we're not duplicating steps/code over and over.
+     *
+     * @param builtToken
+     *            - the token built to reflect the goal of the calling test
+     * @param expectations
+     *            - the expected behavior that we need to validate
+     * @throws Exception
+     */
+    public void genericConfigTest(LibertyServer rs_server, String builder, String location, String name, Expectations expectations) throws Exception {
+
+        String thisMethod = "genericConfigTest";
+        loggingUtils.printMethodName(thisMethod);
+
+        for (List<String> app : getTestAppArray(rs_server)) {
+            standardTestFlow(builder, app.get(0), app.get(1), location, name, expectations);
```

Comment:
Doing `app.get(0)` and `app.get(1)` to get the app root context and app (name?) is pretty unclear. Could the test app array actually be something like a list or set of classes with getters for the actual elements that make up the app/app string?

Reply:
Added a testApp class with getters, constructor...